### ISSUE TITLE
Name layers created using batch_norm() utility

### ISSUE THREAD

**Issue by wuaalb** [Lasagne/Lasagne#670]
This is a _very_ minor issue, but I think it would be nice for layers created using the `batch_norm()` utility to be named. This way all layers will be named when e.g. printing out network structures.

The created `BatchNormLayer` and `NonlinearityLayer` layers could be named by appending a suffix to the name of the wrapped layer, separated by a dot (like in the recurrent layers). In case the `kwargs` argument passed on to the BN layer already contains a name, it could be used to name the `BatchNormLayer` and as a base name for the `NonlinearityLayer`. If neither the wrapped layer is named, nor a name is specified in `kwargs`, the created layers could remain unnamed.

I could make a PR for this.

**Comment by f0k** [comment]
I guess I'd prefer an underscore for the separation instead of a dot (the dot was meant to convey that the wrapped layers are not accessible from the outside, but can be accessed as properties of the recurrent layer), but I'm all for it! Something like `_batchnorm` and `_nonlinearity`? Or just `_bn` and `_nonlin`?
